modules/client/test/integration/client/encoding.test.ts

```diff
@@ -585,7 +585,73 @@ describe("Client", () => {
         );
       }
     });
-    it("Should encode an applyUpdate action", async () => {
+    it("Should encode an applyUpdate action without permissions", async () => {
```

Comment:
please add here also checks that the upgrade and the root permission is properly granted and revoked